HW_27.10/HW_27.10/Controllers/Controller.cs

```diff
@@ -0,0 +1,94 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace HW_27._10.Controllers
+{
+    [ApiController]
+    [Route("book")]
+    public class Controller : ControllerBase
+    {
+        private readonly IConfiguration _config;
+        public Controller(IConfiguration config)
+        {
+            _config = config;
+        }
+
+        [HttpGet("all")]
+        public List<Book> GetAll()
+        {
+            using (StreamReader streamReader = new StreamReader(_config.GetValue<string>("Path1")))
+            {
+                var json = streamReader.ReadToEnd();
+                return JsonConvert.DeserializeObject<List<Book>>(json);
+            }
+        }
+
+        [HttpGet("one/{id}")]
+        public Book GetOne(Guid id)
+        {
+            using (StreamReader streamReader = new StreamReader(_config.GetValue<string>("Path1")))
+            {
+                var json = streamReader.ReadToEnd();
+                var books = JsonConvert.DeserializeObject<List<Book>>(json);
+                foreach (var item in books)
+                {
+                    if (id == item.Id)
+                    {
+                        return item;
+                    }
+                }
+                return null;
+            }
+        }
+
+        [HttpPost("add")]
+        public void Post([FromBody] Book book)
+        {
+            List<Book> books = new List<Book>();
+            var newBook = new Book() { Name = book.Name, Author = book.Author, Id = book.Id, Pages = book.Pages, Popularity = book.Popularity };
```

Comment:
never pass Id for new entity 
always generate in on backend 
getting it in request will cause check for already existing record with this id
but when you generate GUID yourself you can be sure it is unique
![image](https://user-images.githubusercontent.com/58072362/199584496-4108c05e-f9cb-417f-b5a5-be2427ab8dc1.png)